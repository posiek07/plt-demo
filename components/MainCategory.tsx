import React from 'react';
import { StyleSheet, Image, TouchableOpacity, View, Text } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { MainCategoryProps } from '../types';

const MainCategory: React.FC<MainCategoryProps> = ({ title, image, categories }) => {
  const navigation = useNavigation();
  return (
    <TouchableOpacity
      onPress={() => {
        navigation.navigate('SubCategoriesScreen', {
          subCategories: categories,
        });
      }}
      style={styles.card}>
      <Text style={styles.cardTitle}>{title}</Text>
      <View style={styles.cardImageWrapper}>
        <Image style={styles.cardImage} source={{ uri: image }} />
      </View>
    </TouchableOpacity>
  );
};

export default MainCategory;

const styles = StyleSheet.create({
  card: {
    width: '95%',
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#cccccc',
    justifyContent: 'space-between',
  },
  cardImageWrapper: {
    height: 70,
    justifyContent: 'flex-start',
    alignItems: 'flex-end',
    backgroundColor: '#cccccc',
  },
  cardImage: {
    flex: 1,
    backgroundColor: '#cccccc',
    height: 70,
    width: 100,
    resizeMode: 'contain',
  },
  cardTitle: {
    fontSize: 20,
    fontWeight: 'bold',
    paddingLeft: 5,
  },
});
