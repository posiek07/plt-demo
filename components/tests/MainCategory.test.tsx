import React from 'react';
import renderer from 'react-test-renderer';
import MainCategory from '../MainCategory';
jest.mock('@react-navigation/native');

const navigation = [
    {
        name: 'NEW IN',
        img: 'https://cdn-media.prettylittlething.com/studio/mega-menu/desktop/12.11.20/newin.jpg',
        children: [
            {
                name: 'NEW IN',
                categories: ['View all', 'New in today', 'New in this week'],
            },
            {
                name: 'PLT RANGES',
                categories: ['New in plus', 'New in petite', 'New in tall'],
            },
        ],
    },
];
test('renders correctly', () => {
    const tree = renderer
        .create(
            <MainCategory
                title={navigation[0].name}
                image={navigation[0].img}
                categories={navigation[0].children}
            />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
