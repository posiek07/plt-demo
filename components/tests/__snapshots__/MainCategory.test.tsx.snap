// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#cccccc",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "width": "95%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "paddingLeft": 5,
      }
    }
  >
    NEW IN
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#cccccc",
        "height": 70,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn-media.prettylittlething.com/studio/mega-menu/desktop/12.11.20/newin.jpg",
        }
      }
      style={
        Object {
          "backgroundColor": "#cccccc",
          "flex": 1,
          "height": 70,
          "resizeMode": "contain",
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
